x-pack/plugins/enterprise_search/server/utils/fetch_enterprise_search.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import AbortController from 'abort-controller';
+import fetch, { RequestInit, BodyInit } from 'node-fetch';
+
+import { KibanaRequest, Logger } from '@kbn/core/server';
+
+import { ConfigType } from '..';
+
+import { entSearchHttpAgent } from '../lib/enterprise_search_http_agent';
+
+export interface ResponseError {
+  responseStatus: number;
+  responseStatusText: string;
+}
+
+export function isResponseError(resp: unknown): resp is ResponseError {
+  if (typeof resp !== 'object') return false;
+  if (resp === null) return false;
+  if ('responseStatus' in resp && 'responseStatusText' in resp) return true;
+  return false;
+}
+
+export async function fetchEnterpriseSearch<T>(
```

Comment:
I cribbed this pretty much wholesale from [callEnterpriseSearchConfigAPI](https://github.com/elastic/kibana/blob/main/x-pack/plugins/enterprise_search/server/lib/enterprise_search_config_api.ts#L41) as it was the one place I saw we fetch data from ent-search without using the `enterpriseSearchRequestHandler`. I expect we can just delete this when we have a helper to fetch engine from elasticsearch instead of ent-search

Reply:
We have a few other spots where we call enterprise search directly IIRC, though truth be told I couldn't find them when I went looking right now. This will work, but I'd strip out those timeout checks and warnings--those are meant for access checks, not really for regular API calls (you'll note that the enterpriseSearchErrorHandler doesn't use them either).

We could also consider splitting out the actual calling of enterprise search from the enterpriseSeachErrorHandler so we can use that as a separate function.

Reply:
rgr, I'll take a look and remove the unneeded things.